qiskit/pulse/macros.py

```diff
@@ -11,14 +11,29 @@
 # that they have been altered from the originals.
 
 """Module for common pulse programming macros."""
+from __future__ import annotations
 
-from typing import Dict, List, Optional, Union
+from typing import Dict, List, Optional, Union, TYPE_CHECKING
 
 from qiskit.pulse import channels, exceptions, instructions, utils
 from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
 from qiskit.pulse.schedule import Schedule
+from qiskit.utils.deprecation import deprecate_arg
 
 
+if TYPE_CHECKING:
+    from qiskit.transpiler import Target
+
+
+@deprecate_arg(
+    "backend",
+    deprecation_description=(
+        "Depricating ``backendV1`` as the type of measure's `backend` argument."
+    ),
+    additional_msg=("Instead use ``backendV2``as the type of measure's `backend` argument."),
+    since="0.25.0",
```

Comment:
```suggestion
    since="0.24.0",
```
0.25 or 0.24?

Reply:
TBH, I would just drop the `since` field as this is a pending deprecation it's not on any timer and by default isn't user facing so people knowing that we're planning to deprecate this in the future doesn't really have a start date as there is no timer associated with it yet.

That being said I don't think we need this at all. `BackendV1` is still a support interface in Qiskit we haven't marked it as deprecated or anything yet so I would just remove the use of the deprecation decorator here since I'd expect the pulse builder to support both v1 and v2.

Reply:
I deleted the deprecate decorator because "since" field must be provided.

Reply:
@Eric-Arellano here is another case we need to consider :) just fyi

Reply:
Yes, the Sphinx `deprecated` directive expects us to always have a `since`, so that is why it's required.

The docstring for `since` says this:

```
since: The version the deprecation started at. If the deprecation is pending, set
the version to when that started; but later, when switching from pending to
deprecated, update `since` to the new version.
```

> I deleted the deprecate decorator because "since" field must be provided.

I encourage you to still use it and set `pending=True`. That has some benefits:

1. The message will be standardized
2. The pending deprecation will show up in our docs. (It will say its pending, not actually deprecated)
3. It's easy to switch from `pending` to deprecated. You delete `pending=True` and bump `since` to the version it became deprecated.